app-typescript/components/SelectGrid.tsx

```diff
@@ -19,17 +19,21 @@ interface IProps {
     getItems(searchString: string | null): Promise<Array<IItem>>;
     onChange(value: IItem): void;
     itemTemplate: React.ComponentType<{ item: IItem | null }>;
-    triggerTemplate: React.ComponentType<any>;
+    triggerTemplate: React.ComponentType<{onClick: (e: React.SyntheticEvent) => void; }>;
     label: string;
     filterPlaceholder?: string;
 }
 
 interface IState {
     items: Array<IItem>;
     loading: boolean;
-    isOpen: boolean;
 }
 
+const GRID_COLS = 3;
+const GRID_ROWS = 1;
```

Comment:
imo there is not just 1 row, is there? in theory I would expect `PAGE_ITEMS = GRID_COLS * GRID_ROWS`

Reply:
Yep, but I renamed the offsets to `GRID_COLS` & `GRID_ROWS`. Actually they're meant to be `ROW_OFFSET` & `COL_OFFSET`. do you think those names are better?

Reply:
confusion comes because this isn't a 2d array, it's 1d, so in theory there's no rows and cols

Reply:
question is how many rows are actually there, so I would probably set proper value to `GRID_ROWS` so that the `PAGE_SIZE = GRID_COLS * GRID_ROWS` and where you use that `GRID_ROWS` to add/subtract `1` I would just use `1` there, that's not a magic number imo

Reply:
and it's fine to be stored as an array, this should rather describe how it's rendered